src/MarkdownWriter/CommandHelpMarkdownWriter.cs

```diff
@@ -0,0 +1,255 @@
+
+        private void WriteExamples(CommandHelp help)
+        {
+            sb.AppendLine(Constants.ExamplesMdHeader);
+            sb.AppendLine();
+
+            int? totalExamples = help?.Examples?.Count;
+
+            for (int i = 0; i < totalExamples; i++)
+            {
+                sb.Append(help?.Examples?[i].ToExampleItemString(i + 1));
+                sb.AppendLine();
+            }
+        }
+
+        private void WriteParameters(CommandHelp help)
+        {
+            sb.AppendLine(Constants.ParametersMdHeader);
+            sb.AppendLine();
+
+            // Sort the parameter by name before writing
+            help.Parameters?.Sort((u1, u2) => u1.Name.CompareTo(u2.Name));
+
+            if (help.Parameters?.Count > 0)
+            {
+                foreach (Parameter param in help.Parameters)
+                {
+                    string paramString = param.ToParameterString();
+
+                    if (!string.IsNullOrEmpty(paramString))
+                    {
+                        sb.AppendLine(paramString);
+                        sb.AppendLine();
+                    }
+                }
+
+                if (help.HasCmdletBinding)
+                {
+                    sb.AppendLine(Constants.CommonParameters);
+                }
+            }
+        }
+
+        private void WriteInputsOutputs(List<InputOutput> inputsoutputs, string header)
+        {
+            sb.AppendLine(header);
+            sb.AppendLine();
+
+            if (inputsoutputs is null)
+            {
+                return;
+            }
+
+            foreach (var item in inputsoutputs)
+            {
+                sb.Append(item.ToInputOutputString());
+            }
+        }
+
+        private void WriteNotes(CommandHelp help)
+        {
+            sb.AppendLine(Constants.NotesMdHeader);
+            sb.AppendLine();
+            sb.AppendLine(help.Notes);
+            sb.AppendLine();
+        }
+
+        private void WriteRelatedLinks(CommandHelp help)
+        {
+            sb.AppendLine(Constants.RelatedLinksMdHeader);
+            sb.AppendLine();
+
+            if (help.RelatedLinks?.Count > 0)
+            {
+                foreach (var link in help.RelatedLinks)
+                {
+                    sb.AppendLine(link.ToRelatedLinksString());
+                    sb.AppendLine();
+                }
+            }
+            else
+            {
+                sb.AppendLine(Constants.FillInRelatedLinks);
+                sb.AppendLine();
+            }
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    Constants.StringBuilderPool.Return(sb);
+                }
+
+                disposedValue = true;
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
```

Comment:
Do you expect `CommandHelpMarkdownWriter` to be inherited by another type? If not, then add `sealed` to the type declaration, and then just move the logic in `Dispose(bool disposing)` here and remove the `GC.SuppressFinalize` call. You don't even need the `disposedValue` field unless you are checking it somewhere else in the code and will throw a `ObjectDisposedException`.

Reply:
fixed.